### PR TITLE
WebUSB cleanup

### PR DESCRIPTION
**Pull request by pelikhan**
Cleaning up the PacketIO story. 

- [x] removed duplicate layer in micro:bit and centralize all exception/error handling in pxt
- [x] move PacketIO interface to separate file
- [x] introduce new interface, PacketIOWrapper that can be HF2 (arcade) or DAP (micro:bit)

## micro:bit

* live: https://makecode.microbit.org/app/7ffa4d587eb1a9cbeedba7c23918d39d6025c997-5b40138d06
* PR: https://github.com/microsoft/pxt-microbit/pull/2794

#### macos, windows 10

- [x] deploy
- [x] console logging
- [x] auto reconnect

### winrt:

- [x] deploy
- [x] console logging

### electron:

- [x] deploy to drives

### webusb:

- [x] deploy
- [x] console logging
- [x] auto reconnect

### CODE REVIEW

File: pxtlib/webusb.ts [hunk 1]
```diff
@@ -320,54 +391,19 @@ namespace pxt.usb {
...
-        packets: USBIsochronousOutTransferPacket[];
-    }
-
     export function pairAsync(): Promise<void> {
         return ((navigator as any).usb.requestDevice({
             filters: filters
         }) as Promise<USBDevice>).then(dev => {
             // try connecting to it
-            return mkPacketIOAsync()
+            return pxt.usb.mkPacketIOAsync()
```
Comment (pelikhan) [inline pxtlib/webusb.ts:406]: needs review

File: webapp/src/cmds.ts [hunk 2]
```diff
@@ -143,31 +147,55 @@ function nativeHostSaveCoreAsync(resp: pxtc.CompileResult): Promise<void> {
...
-                pxt.tickEvent("hid.flash.devicenotfound");
-                return d.reportDeviceNotFoundAsync(troubleshootDoc, resp);
-            } else {
+    let isRetry = false;
+    return deployAsync();
+
+    function deployAsync(): Promise<void> {
+        return pxt.packetio.initAsync(isRetry)
+            .then(dev => dev.reflashAsync(resp)
+                .then(() => dev.disconnectAsync())
```
Comment (pelikhan) [inline webapp/src/cmds.ts:162]: review this

File: webapp/src/data.ts [hunk 3]
```diff
@@ -275,6 +275,7 @@ export function stripProtocol(path: string) {
 }
 
 export function invalidate(prefix: string) {
+    prefix = prefix.replace(/:\*$/, ':'); // remove trailing "*";
```
Comment (pelikhan) [inline webapp/src/data.ts:278]: all headers:* were wrong

**Comment by pelikhan** [comment]
@mmoskal ready for review.

File: webapp/src/data.ts (hunk 3 above)
Comment (mmoskal) [reply webapp/src/data.ts:278]: oh wow!

Comment (pelikhan) [reply webapp/src/data.ts:278]: Yup, we didn’t invalidate tons of stuff ?!??!

**Pull request merged**